src/components/inscription/Form.js

```diff
@@ -54,29 +54,33 @@ class Form extends React.Component {
   onSubmit = ({ formData }) => {
     const url =
       this.state.typeMandataire === "ti" ? "/inscription/tis" : "/inscription/mandataires";
-    this.setState({ status: "loading", formData }, () => {
-      apiFetch(url, {
-        method: "POST",
-        body: JSON.stringify({
-          ...formData,
-          etablissement: formData.etablissement || "",
-          tis: this.state.tis,
-          type: this.state.typeMandataire,
-          username: formData.username.toLowerCase().trim()
+    if (formData.code_postal.match(/^(([0-8][0-9])|(9[0-5])|(2[AB]))[0-9]{3}$/)) {
+      this.setState({ status: "loading", formData }, () => {
+        apiFetch(url, {
+          method: "POST",
+          body: JSON.stringify({
+            ...formData,
+            etablissement: formData.etablissement || "",
+            tis: this.state.tis,
+            type: this.state.typeMandataire,
+            username: formData.username.toLowerCase().trim()
+          })
         })
-      })
-        .then(json => {
-          if (json.success === false) {
-            throw new Error();
-          }
-          // piwik.push(["trackEvent", "Inscription success", formData.type]);
-          Router.push("/inscription-done");
-        })
-        .catch(() => {
-          // piwik.push(["trackEvent", "Inscription error", formData.type]);
-          this.setState({ status: "error" });
-        });
-    });
+          .then(json => {
+            if (json.success === false) {
+              throw new Error();
+            }
+            // piwik.push(["trackEvent", "Inscription success", formData.type]);
+            Router.push("/inscription-done");
+          })
+          .catch(() => {
+            // piwik.push(["trackEvent", "Inscription error", formData.type]);
+            this.setState({ status: "error" });
```

Comment:
les erreurs sont-elles gérées à l'affichage ?

Reply:
?

Reply:
je ne vois pas de gestion de status==="error" dans le render de ce composant donc est-ce que l'usager est notifié quand il y a une erreur ?

Reply:
it is !